feature/google/src/main/java/com/revenuecat/purchases/google/findBestOffer.kt

```diff
@@ -0,0 +1,115 @@
+@file:Suppress("MaximumLineLength", "MaxLineLength")
+package com.revenuecat.purchases.google
+
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.PricingPhase
+import com.revenuecat.purchases.models.RecurrenceMode
+
+fun List<GoogleSubscriptionOption>.findBestOffer(): GoogleSubscriptionOption? {
+    val basePlan = this.firstOrNull { it.isBasePlan } ?: return null
+
+    val validOffers = this
+        .filter { !it.isBasePlan }
+        .filter { !it.tags.contains("rc-ignore-best-offer") }
+
+    return findLongestFreeTrial(validOffers) ?: findBestSavingsOffer(basePlan, validOffers) ?: basePlan
```

Comment:
Would a sorting algorithm be easier to make and understand? The comparator of two `GoogleSubscriptionOption` would be something like:

```
compare(Option a, Option b) {
  if (a.freeTrialLength() != b.freeTrialLength()) return b-a;
  if (a.introPrice() != b.introPrice()) return a-b; //for introprice the default value should be MAX_INT for this to work
}
```
I assume something like this should work if we sort the options...

Reply:
Ohhh, this does look like it could work too! But... my mind does not process comparators very well at all 😬